### PR TITLE
Move out JsonldContextGenerator

### PR DESCRIPTION
**Pull request by whikloj**
Remove Web tests and update FedoraResource JsonLd controller

Partially resolves: https://github.com/Islandora-CLAW/CLAW/issues/596
Relies on: https://github.com/Islandora-CLAW/jsonld/pull/22

# What does this Pull Request do?

Moves out the JsonLdContextGenerator to the jsonld module.

# How should this be tested?

See steps on associated PR https://github.com/Islandora-CLAW/jsonld/pull/22

# Interested parties
@Islandora-CLAW/committers @Natkeeran

### CODE REVIEW

File: src/Tests/Web/JsonldContextGeneratorWebTest.php [hunk 1]
```diff
@@ -1,96 +0,0 @@
...
-use Drupal\Core\Url;
-use Drupal\islandora\Entity\FedoraResourceType;
-use Drupal\rdf\Entity\RdfMapping;
-
-/**
- * Implements WEB tests for Context routing response in various scenarios.
- *
- * @group islandora
- */
-class JsonldContextGeneratorWebTest extends IslandoraWebTestBase {
```
Comment (dannylamb) [inline src/Tests/Web/JsonldContextGeneratorWebTest.php]: This web test should stay with the controller in the Islandora module.

Comment (whikloj) [inline src/Tests/Web/JsonldContextGeneratorWebTest.php]: Fair point, should it be in `src/Tests/` or `tests/src/Functional` ?

Comment (whikloj) [inline src/Tests/Web/JsonldContextGeneratorWebTest.php]: or maybe `tests/src/Web` ?

Just seems wrong inside the `src/` folder

Comment (dannylamb) [inline src/Tests/Web/JsonldContextGeneratorWebTest.php]: I think @DiegoPino had specific reasons for putting it where it is.  I'd prefer it live in the `tests/src` folder somewhere, but iirc there were issues with getting it to run through the browser.  Maybe he can speak to it more.

Comment (whikloj) [inline src/Tests/Web/JsonldContextGeneratorWebTest.php]: Actually, @DiegoPino was right and depending on how you read the Drupal docs, our current Kernel Tests are in the wrong place.
https://github.com/Islandora-CLAW/CLAW/issues/621

**Comment by DiegoPino** [comment]
@whikloj right now they are in https://github.com/Islandora-CLAW/islandora/tree/8.x-1.x/src/Tests/Web  because they are using the legacy [WebTest framework ](https://api.drupal.org/api/drupal/core%21modules%21simpletest%21src%21WebTestBase.php/class/WebTestBase/8.2.x). Since they are not PHPUnit test they have this different folder structure (looking at other modules + not being able to make them run if this was not the place, made me do that thing). 
Drupal\simpletest\WebTestBase folder

Will be at least supported until 8.4.x and in still used in at least in 400 different places in Drupal core. If you want them in /tests/src/Functional then you will have to transform them to BrowserTests https://www.drupal.org/docs/8/testing/types-of-tests-in-drupal-8 which can be a good ticket since there are plans to move eventually everything there

**Comment by whikloj** [comment]
I'm rebasing this PR so I can go back and keep the files and maintain commit history.

**Comment by whikloj** [comment]
This is going to fail until https://github.com/Islandora-CLAW/jsonld/pull/22 is merged to add the `jsonld.contextgenerator` service.

**Comment by dannylamb** [comment]
Restarting Travis since https://github.com/Islandora-CLAW/jsonld/pull/22 has landed

**Pull request merged**